File: 3871-minimum-deletions-for-at-most-k-distinct-characters/minimum-deletions-for-at-most-k-distinct-characters.py
class Solution:
    def minDeletion(self, s: str, k: int) -> int:
        freq = Counter(s)

        if len(freq) <= k:
            return 0
        
        frequencies = sorted(freq.values())
        deletions_needed = len(freq) - k

        return sum(frequencies[:deletions_needed])

File: 3871-minimum-deletions-for-at-most-k-distinct-characters/minimum-deletions-for-at-most-k-distinct-characters.js
/**
 * @param {string} s
 * @param {number} k
 * @return {number}
 */
var minDeletion = function(s, k) {
    const freq = {};
    let result = 0;
    for (const char of s) {
        freq[char] = (freq[char] || 0) + 1;
        }
    const frequencies = Object.values(freq);
    frequencies.sort((a, b) => a - b);
    while (frequencies.length > k) {
        result += frequencies[0];
        frequencies.shift();
    }
    return result;
};